modules/apps/commerce/commerce-channel-web/test/js/commerce_marketplace_payment_method/index.spec.tsx

```diff
@@ -0,0 +1,428 @@
+
+		await act(async () => jest.runAllTimers());
+
+		expect(queryByText(productRequestRsponse.items[0].name)).toBeTruthy();
+
+		fireEvent.click(
+			queryByText(
+				productRequestRsponse.items[0].name
+			) as HTMLButtonElement
+		);
+
+		expect(fetch.mock.calls.length).toBe(2);
+		expect(queryByText('back-to-list')).toBeTruthy();
+	});
+
+	it('renders autorized with apps and click on cancel button', async () => {
+		fetch
+			.mockResponseOnce(JSON.stringify(marketplaceSettings))
+			.mockResponseOnce(JSON.stringify(productRequestRsponse));
+
+		const {queryByText} = await render(<CommerceChannelAddPaymentMethod />);
+
+		const addButton = queryByText('add');
+
+		await fireEvent.click(addButton as HTMLButtonElement);
+
+		await act(async () => jest.runAllTimers());
+
+		expect(queryByText(productRequestRsponse.items[0].name)).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('cancel') as HTMLButtonElement
+				)
+		);
+
+		expect(fetch.mock.calls.length).toBe(2);
+
+		expect(queryByText(productRequestRsponse.items[0].name)).toBeTruthy();
+	});
+
+	it('renders authorized with apps and click install failed', async () => {
+		const createCart = {
+			cartItems: [
+				{
+					id: 37062624,
+				},
+			],
+			id: 37062623,
+		};
+
+		fetch
+			.mockResponseOnce(JSON.stringify(marketplaceSettings))
+			.mockResponseOnce(JSON.stringify(productRequestRsponse))
+			.mockResponseOnce(JSON.stringify(createCart))
+			.mockRejectOnce(JSON.stringify({ok: false}));
+
+		const {queryByText} = render(<CommerceChannelAddPaymentMethod />);
+
+		const addButton = queryByText('add');
+
+		await fireEvent.click(addButton as HTMLButtonElement);
+
+		await act(async () => jest.runAllTimers());
+
+		expect(queryByText('Map for Objects')).toBeTruthy();
+
+		await fireEvent.click(
+			queryByText('Map for Objects') as HTMLButtonElement
+		);
+
+		expect(queryByText('back-to-list')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(queryByText('confirmation-required')).toBeTruthy();
+
+		await act(
+			async () =>
+				await fireEvent.click(
+					queryByText('install') as HTMLButtonElement
+				)
+		);
+
+		expect(fetch.mock.calls.length).toBe(4);
+
+		expect(queryByText('there-was-an-unknown-error')).toBeTruthy();
+	});
+
+	it('renders authorized with apps and click install success', async () => {
+		const mockCartResponse = {
+			cartItems: [
+				{
+					id: 123,
+				},
+			],
+			id: 123,
+		};
+
+		fetch
+			.mockResponseOnce(JSON.stringify(marketplaceSettings))
+			.mockResponseOnce(JSON.stringify(productRequestRsponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify(mockCartResponse))
+			.mockResponseOnce(JSON.stringify({ok: true}));
```

Comment:
Keep it consistent 

previously you called "createCart" and now "mockCartResponse" for the same thing, variable

Inline this (remove the variable) or add to the top of the file if a generic name like "cartResponseMock"